pkg/cloudfabric-controller/controllerframework/controller_framework.go

```diff
@@ -632,9 +440,18 @@ func (c *ControllerBase) ReportHealth() {
 	}
 }
 
-func generateControllerName() string {
-	uid := uuid.NewUUID()
-	return fmt.Sprintf("%s-%v", controllerInstanceNamePrefix, uid)
+func generateControllerName(existedInstanceMap map[string]v1.ControllerInstance) string {
+	for {
+		uid := uuid.NewUUID()
+		name := fmt.Sprintf("%s-%v", controllerInstanceNamePrefix, uid)
+		_, isExist := existedInstanceMap[name]
```

Comment:
Retry should solve the problem. The chance that a unique name can not be created after a few retrials is almost zero.

Reply:
> Retry should solve the problem. The chance that a unique name can not be created after a few retrials is almost zero.

This is retry. Right?